networking.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: page
+title: Making Networking Requests in Flutter
+permalink: /networking/
+---
+
+* TOC Placeholder
+{:toc}
+
+## Using the `http` package
+
+Flutter supports the the [http package][http], version `0.11.3+12` or later.
+
+Start by declaring a dependency by adding an http line under `dependencies` in
+`pubspec.yaml`:
+
+```
+dependencies:
+  ...
+  http: '>=0.11.3+12'
+```
+
+## Making requests
+
+You can create an HTTP [Client][client] with the `Client` constructor:
+
+<!-- skip -->
+```dart
+import 'package:http/http.dart' as http;
+
+http.Client _httpClient = new http.Client();
+```
+
+The client supports common HTTP operations:
+
+* **HTTP GET:** Use [`get`][get] for a general request, [`read`][read] for a
+ request returning a string, or [`readbytes`][readbytes] for a request
+ returning bytes.
+
+* **HTTP POST:** Use [`post`][post] for a general post.
+
+Note that the API uses [Dart
+Futures](https://www.dartlang.org/tutorials/language/futures) in the return
+values. These are most easily used with the `then()` methods like this:
+
+<!-- skip -->
+```dart
+_httpClient.post(url, body: {"name": "doodle", "color": "blue"})
+    .then((response) {
+  print("Response status: ${response.statusCode}");
+});
+```
+
+## Decoding and encoding JSON
+
+This is supported via the [`dart:convert`](https://docs.flutter.io/flutter/dart-convert/dart-convert-library.html) library.
+
+To decode/parse the JSON string response into a Map:
+
+<!-- skip -->
+```dart
+Map data = JSON.decode(response.body);
+// Assume the response body is something like: ["foo", { "bar": 499 }]
+int barValue = data[1]['bar']; // barValue is set to 499
+```
+
+To encode JSON simply pass in a `Map` with the values:
```

Comment:
Maybe say "pass a Map or List of Maps, Lists, and string, boolean, or number literals"